directories.go

```diff
@@ -0,0 +1,39 @@
+package maddy
+
+import (
+	"os"
+)
+
+var (
+	defaultConfigDirectory  = "/etc/maddy"
+	defaultStateDirectory   = "/var/lib/maddy"
+	defaultLibexecDirectory = "/usr/libexec/maddy"
+)
+
+func ConfigDirectory() string {
+	return defaultConfigDirectory
+}
+
+func StateDirectory(globals map[string]interface{}) string {
+	if os.Getenv("MADDYSTATE") != "" {
```

Comment:
Ah, one more thing: this is racy. Not a big deal, but something like this would be cleaner:

```go
if dir := os.Getenv("MADDYSTATE"); dir != "" {
	return dir
}
```

Reply:
Is it possible to change the environment of the running process? [Googls says no](https://stackoverflow.com/a/8589266).

Reply:
https://golang.org/pkg/os/#Setenv

Reply:
Do we have any reason to change it from maddy? I don't so. Anyway, will change it because I don't like duplicated envvar name.